domain/port/state/watcher_test.go

```diff
@@ -110,38 +111,30 @@ FROM unit_endpoint
 *   - https is open on endpoint 2 on unit 2
  */
 
-func (s *watcherSuite) TestGetMachinesForUnitEndpoints(c *gc.C) {
+func (s *watcherSuite) TestGetMachinesForEndpoints(c *gc.C) {
 	st := NewState(s.TxnRunnerFactory())
 	ctx := context.Background()
 	endpointUUIDs, endpointToUUIDMap := s.initialiseOpenPorts(c, st)
 
-	machineUUIDsForUnitEndpoints, err := st.GetMachineNamesForUnitEndpoints(ctx, endpointUUIDs)
+	machineUUIDsForEndpoint, err := st.GetMachineNamesForUnitEndpoints(ctx, endpointUUIDs)
 	c.Assert(err, jc.ErrorIsNil)
-	c.Check(machineUUIDsForUnitEndpoints, jc.SameContents, []machine.Name{"0", "1"})
+	c.Check(machineUUIDsForEndpoint, jc.SameContents, []machine.Name{"0", "1"})
 
-	machineUUIDsForUnitEndpoints, err = st.GetMachineNamesForUnitEndpoints(ctx, []string{endpointToUUIDMap["ep0"]})
+	machineUUIDsForEndpoint, err = st.GetMachineNamesForUnitEndpoints(ctx, []string{endpointToUUIDMap["ep0"]})
 	c.Assert(err, jc.ErrorIsNil)
-	c.Check(machineUUIDsForUnitEndpoints, jc.DeepEquals, []machine.Name{"0"})
-
-	machineUUIDsForUnitEndpoints, err = st.GetMachineNamesForUnitEndpoints(ctx, []string{endpointToUUIDMap["ep0"], endpointToUUIDMap["ep1"]})
-	c.Assert(err, jc.ErrorIsNil)
-	c.Check(machineUUIDsForUnitEndpoints, jc.DeepEquals, []machine.Name{"0"})
+	c.Check(machineUUIDsForEndpoint, jc.DeepEquals, []machine.Name{"0"})
 }
 
-func (s *watcherSuite) TestGetApplicationForUnitEndpoints(c *gc.C) {
+func (s *watcherSuite) TestFilterEndpointForApplication(c *gc.C) {
 	st := NewState(s.TxnRunnerFactory())
 	ctx := context.Background()
 	endpointUUIDs, endpointToUUIDMap := s.initialiseOpenPorts(c, st)
 
-	appNamesForUnitEndpoints, err := st.GetApplicationNamesForUnitEndpoints(ctx, endpointUUIDs)
-	c.Assert(err, jc.ErrorIsNil)
-	c.Check(appNamesForUnitEndpoints, jc.SameContents, []string{appNames[0], appNames[1]})
-
-	appNamesForUnitEndpoints, err = st.GetApplicationNamesForUnitEndpoints(ctx, []string{endpointToUUIDMap["ep0"]})
+	filteredEndpointUUIDs, err := st.FilterEndpointsForApplication(ctx, s.appUUIDs[0], endpointUUIDs)
```

Comment:
**suggestion:** both changed test in this file aren't very readable without context (we need to check watcher_test, which is not documented, so we have to switch file and read some code to check the asserts.) 

A good think would be to not have to do that, either:

- by having an initialisation method which make it explicit, in the test, what is the topology (and just hide the boilerplate)
- Add some comment (more prone to be broken)
- Add some document (more prone to be broken, a bit less convenient.)

Reply:
We do have some comments have describing the context. Unfortunately they aren't visible in the diff

https://github.com/juju/juju/blob/6166c9abb53c6bfec5f57bcbd5f032032e20d881/domain/port/watcher_test.go#L147-L156
and
https://github.com/juju/juju/blob/6166c9abb53c6bfec5f57bcbd5f032032e20d881/domain/port/state/watcher_test.go#L97-L111

Reply:
But I do agree, encoding it in code itself is best